src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -162,21 +202,33 @@ function useChatUtilsHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatUtilsHook {
       channelData.channelPicture = channelImage;
       channelData.name = channelName;
 
-      await channelData.save(localDb);
-    }
-
-    const promise = originalMembers?.map((member) => {
-      return new Promise((resolve, reject) => {
-        const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-        try {
-          user.saveOrUpdateIfExists(localDb).then(() => resolve(null));
-        } catch (e) {
-          reject(e);
+      queue.addPriorityJob({
+        operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveChannelList,
+        id: channel?.id,
+        priority: QueueJobPriority.MEDIUM,
+        task: async () => {
+          channelData.saveIfLatest(localDb);
         }
       });
-    });
+    }
 
-    await Promise.all(promise).catch((e) => console.log('saveChannelDetail error', e));
+    originalMembers?.map((member) => {
+      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+      try {
+        queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveUserMember,
+          id: `${channel?.name}-${member?.user?.username}`,
+          priority: QueueJobPriority.MEDIUM,
+          task: async () => {
+            user.saveOrUpdateIfExists(localDb);
+          }
+        });
+      } catch (e) {
+        console.log('error on save user from channel detail fetch', e);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, consider adding error handling within each task to ensure that one failing operation doesn't stop the entire queue.
```diff
        queue.addPriorityJob({
          operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveUserMember,
          id: `${channel?.name}-${member?.user?.username}`,
          priority: QueueJobPriority.MEDIUM,
          task: async () => {
            try {
              user.saveOrUpdateIfExists(localDb);
            } catch (error) {
              console.log('Error saving or updating user:', error);
            }
          }
        });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->